### ISSUE TITLE
Inconsistencies with status and severity

### ISSUE THREAD

**Issue by NomAnor** [pyepics/pyepics#56]
Hi,
I'm monitoring some records and I was wondering why I got a status value of 1 in the callback while a caget on the command line gave me no alarm status. When creating a PV, `form='time'` and `form='ctrl'` give different values (with different meanings) in the pv.status attribute.

After digging around in the code I found the problem. In the file ca.py in the function _onMonitorEvent the status keyword is assigned the value of the status argument (Line 465). According to the EPICS Channel Access manual (http://www.aps.anl.gov/epics/base/R3-14/12-docs/CAref.html#ca_add_event) this is the status of the call (1 for sucess in caerr.h). The pyepics manual (http://pyepics.github.io/pyepics/pv.html#attributes) states that 1 is the ok value but this is not the alarm status of the PV.

Later in the function if someone uses `form='time'`, the status keyword is overwritten with the record alarm status value (kwds['status'] = tmpv.status). This is what I would expect in all circumstances.

I propose to change line 465 to
`kwds = {'ftype':args.type, 'count':args.count, 'chid':args.chid, 'pvname': pvname}`
and line 475 to:
`for attr in dbr.ctrl_limits + ('precision', 'units', 'status', 'severity'):`
And change the manual accordingly.

The pv.status and pv.severity attributes (and the keywords in the callbacks) should always be the alarm values.

**Comment by newville** [comment]
@NomAnor  Good point.  I have to admit rarely paying attention to status and severity fields.  What *is* `args.status` holding if not, you know, "status"?  It's described as
"ECA_XXX status of the requested op from the server".   
I'm not sure I really understand why (when) that is different from "status"?

Anyway, I'd be OK with your proposed changes.   Would you like to make a PR for the changes to code and doc?

**Comment by klauer** [comment]
The status returned is the status of the call - `ECA_NORMAL` and such, but the status of the PV is alarm status/severity. Alarm information comes in all but the primitive/native type requests (so not `DBR_FLOAT` but `DBR_STS_FLOAT`, `DBR_TIME_FLOAT`, `DBR_CTRL_FLOAT`.)

The suggested changes sound good.

**Comment by newville** [comment]
@NomAnor @klauer  OK, that `args.status` doesn't sound highly useful to me (like, the monitor got called.  Is it worth Should we preserve the value of `args.status` someplace, perhaps 'monitor_status'?    We're not saving this for all data type variants, but perhaps we should?   I'm fine either way.

**Comment by NomAnor** [comment]
I'm not sure how to handle the `args.status` value correctly. As a user I don't want the callback to be invoked when the data of the PV is not defined so the function ignores any bad values. I use the connection callback to know if the record is connected and the data reliable.

**Comment by klauer** [comment]
I think it's a good point that the connection callbacks are mostly what people would care about. However, there are a few other status codes that could potentially be useful for debugging purposes. It's not impossible that someone is currently relying on having access to that.

This issue prompted me to check what happens with `get` callbacks again. It appears in the case of an `_onGetEvent`, all of the additional metadata is ignored - only the unpacked value is stored in the `get_cache`. That is, you could have a `DBR_TIME_FLOAT` without a monitor configured which would never get any alarm status or server timestamps back to the `epics.PV` instance... Perhaps it's a topic for another issue, but it's definitely related.

**Comment by newville** [comment]
@NomAnor @klauer I agree that more consistent handling of the raw `args` fields would be good.  

For `_onMonitorEvent()`, one approach could be that `args.status` could be saved as  kws[`monitor_status`] and passed on to the user-defined callback.   I don't have a strong preference for that over #57, where `_onMonitorEvent()` returns without running the user-defined callback if `args.status` is not `ECA_NORMAL`.   I suspect #57 is just fine (and it fixes the more important problem of the wrong status being used, sometimes).

**Comment by klauer** [comment]
This should be largely resolved after #135. If you disagree, I'd recommend opening a new, targeted issue based on how the above discussion went.
